assets/src/components/site-scan-results/site-scan-source-list-item.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { AMPNotice, NOTICE_TYPE_PLAIN, NOTICE_SIZE_SMALL } from '../amp-notice';
+import { SiteScanSourcesDetail } from './site-scan-sources-detail';
+
+/**
+ * Site Scan sources list item component.
+ *
+ * @param {Object} props                         Component props.
+ * @param {string} props.author                  Plugin/Theme author.
+ * @param {string} props.name                    Plugin/Theme name.
+ * @param {string} props.slug                    Plugin/Theme slug.
+ * @param {string} props.status                  Current status of Plugin/Theme.
+ * @param {string} props.version                 Plugin/Theme version.
+ * @param {string} props.inactiveSourceNotice    Message to show next to an inactive source.
+ * @param {string} props.uninstalledSourceNotice Message to show next to an uninstalled source.
+ */
+export function SiteScanSourcesListItem( {
+	author,
+	name,
+	slug,
+	status,
+	version,
+	inactiveSourceNotice,
+	uninstalledSourceNotice,
+} ) {
+	const [ hasOpened, setHasOpened ] = useState( false );
+	const handleKey = ( { key } ) => {
+		if ( [ ' ', 'return' ].includes( key ) ) {
+			setHasOpened( true );
+		}
+	};
+
+	return (
+		// eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
+		<details
+			open={ false }
+			onKeyDown={ handleKey }
+			onClick={ () => {
+				setHasOpened( true );
+			} }
+		>
+			<summary>
+				<span className="dashicons dashicons-controls-play" />
+				{ name && (
+					<span className={ classnames( 'site-scan-results__source-name', {
+						'site-scan-results__source-name--inactive': [ 'inactive', 'uninstalled' ].includes( status ),
+					} ) }>
+						{ name }
+					</span>
+				) }
+				{ ! name && slug && (
+					<code className="site-scan-results__source-slug">
+						{ slug }
+					</code>
+				) }
+				{ status === 'active' ? (
+					<>
+						{ author && (
+							<span className="site-scan-results__source-author">
+								{ sprintf(
+									/* translators: %s is an author name. */
+									__( 'by %s', 'amp' ),
+									author,
+								) }
+							</span>
+						) }
+						{ version && (
+							<span className="site-scan-results__source-version">
+								{ sprintf(
+									/* translators: %s is a version number. */
+									__( 'Version %s', 'amp' ),
+									version,
+								) }
+							</span>
+						) }
+					</>
+				) : (
+					<AMPNotice
+						className="site-scan-results__source-notice"
+						type={ NOTICE_TYPE_PLAIN }
+						size={ NOTICE_SIZE_SMALL }
```

Comment:
This may be a limitation of IntelliJ/WebStorm but I get a inspection warning here:

<img width="529" alt="image" src="https://user-images.githubusercontent.com/134745/157325741-db9fb9a0-0f58-469a-b54a-160be1c4d13e.png">

It seems when `AMPNotice.propTypes` is set to include `PropTypes.oneOf( [ NOTICE_SIZE_LARGE, NOTICE_SIZE_SMALL ] )` that it isn't able to resolve the reference.

Reply:
Strange, I'm getting no inspection warning in PHPStorm for this code fragment:

<img width="514" alt="Screenshot 2022-03-08 at 23 11 29" src="https://user-images.githubusercontent.com/478735/157334481-41d60e9c-3e5a-4f28-8a4f-6abbfaee5deb.png">